wallet/src/de/schildbach/wallet/transactions/WalletMostRecentTransactionsObserver.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.transactions
+
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.util.ThrottlingWalletChangeListener
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import org.bitcoinj.core.Coin
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.utils.Threading
+import org.bitcoinj.wallet.Wallet
+
+@ExperimentalCoroutinesApi
+class WalletMostRecentTransactionsObserver(private val wallet: Wallet) {
```

Comment:
I wonder if we need a separate class for this. In CrowdNode, there is `WalletTransactionObserver`, and if we add a parameter in there for turning off the `confidenceChanged` emitting, we should have something similar to this.

Perhaps later we can write a bunch of tests to make sure nothing breaks and then try to combine them.

Reply:
I was wondering the same.  Some of the code is similar to the wallet balance observer also.